inference-engine/src/readers/onnx_reader/ie_onnx_reader.hpp

```diff
@@ -39,6 +44,11 @@ class ONNXReader: public IReader {
     std::vector<std::string> getDataFileExtensions() const override {
         return {};
     }
+
+    private:
+        // m_manager should be released after m_onnx_fe
+        ngraph::frontend::FrontEndManager m_manager;
+        ngraph::frontend::FrontEnd::Ptr m_onnx_fe;
```

Comment:
The is time dependency of the lifecycle of `FrontEndManager` and `FrontEnd`.
'FrontEnd` must be released before `FrontEndManager`. Otherwise, the segfault is thrown.
@ilya-lavrenov @ilyachur Do you have ideas on how to deal with it?
The final solution should be applied probably to `FrontEndManager`, but it is also worth considering if `FrontEndManager` should be ctor argument of Readers (to have only one instance of `FrontEndManager` and to preserve correct lifetime of FE objects).

Reply:
I propose to remove ONNX reader and use ONNX frontend instead.

Reply:
To remove reader we need to have IR FE which is in progress https://github.com/openvinotoolkit/openvino/pull/7152
@mbencer lets meet in offline to discuss next steps.

Reply:
@GlebKazantaev As I understand the change which I can make in this PR is removing only ONNX Reader (IR readers can still exist).
The is already code in `ReadNetwork` (added by PDPD guys) which run frontends execution anyway (https://github.com/openvinotoolkit/openvino/blob/e46f6ce573cda3433189b174584aea6f262b67cf/inference-engine/src/inference_engine/src/ie_network_reader.cpp#L243)
But sure, lets meet offline ;)